velox/exec/RowContainer.cpp

```diff
@@ -368,7 +370,7 @@ void RowContainer::freeVariableWidthFields(folly::Range<char**> rows) {
             if (!view.isInline()) {
               stringAllocator_->free(
                   HashStringAllocator::headerOf(view.data()));
-              if (checkFree_) {
+              if (checkFree_ || resetFields) {
```

Comment:
Wondering how does this work for ROW, ARRAY and MAP. These do not store StringView in row, do they?

Reply:
I think you are right. Isn't it confusing to use StringView though? It suggest a value of type VARCHAR. Wondering if std::string_view would be less confusing.

Reply:
I agree, plus the optimizations of StringView doesn't apply to complex types.
I created a PR to address this, please take a look. Thanks! https://github.com/facebookincubator/velox/pull/7349